File: apps/write-8-minutes/ios/Anky/Features/Painting/UnveilCeremonyView.swift
import SwiftUI

/// The level-up ceremony.
///
/// The frame never moves — PaintingFrameMetrics places it exactly where the
/// main screen holds it; the world darkens and blooms around it. Sequence:
/// final strokes → held breath → candlelit aubergine → WELCOME TO LEVEL {N}
/// → the 8-second glimpse (bloom, breath, recede) → ghost Begin → drain.
struct UnveilCeremonyView: View {
    let level: Int
    let coordinator: LevelPaintingCoordinator
    let onFinished: () -> Void

    private enum Beat {
        case finalStrokes
        case heldBreath
        case darkening
        case title
        case waitingForGlimpse
        case glimpseBloom
        case glimpseHold
        case glimpseRecede
        case begin
        case drain
    }

    @State private var beat: Beat = .finalStrokes
    @State private var completedAssets: PaintingRevealAssets?
    @State private var completedTitle = ""
    @State private var glimpseAssets: PaintingRevealAssets?
    @State private var completedProgress: Double = 0.72
    @State private var glimpseProgress: Double = 0
    @State private var darkness: Double = 0
    @State private var titleOpacity: Double = 0
    @State private var beginOpacity: Double = 0
    @State private var glowOvershoot: Double = 0
    @State private var contentOpacity: Double = 1
    @State private var theme = LevelTheme.fallback
    @State private var generationExcerpts: [String] = []

    var body: some View {
        GeometryReader { geometry in
            let frame = PaintingFrameMetrics.frameRect(in: geometry.size)

            ZStack {
                // The room: warm paper below, living aubergine darkness above.
                LazureWall(mood: theme.wallMood)
                aubergineDarkness
                    .opacity(darkness)

                if beat == .waitingForGlimpse {
                    PaintingGenerationWaitView(excerpts: generationExcerpts)
                        .transition(.opacity)
                }

                // The never-moving frame.
                Group {
                    if showsGlimpsePainting, let glimpseAssets {
                        PaintingView(
                            assets: glimpseAssets,
                            progress: glimpseProgress,
                            glowTint: theme.glowTint,
                            glowStrength: 1 + glowOvershoot
                        )
                    } else if let completedAssets {
                        PaintingView(
                            assets: completedAssets,
                            progress: completedProgress,
                            glowTint: theme.glowTint,
                            glowStrength: 1 + glowOvershoot
                        )
                    }
                }
                .frame(width: frame.width, height: frame.height)
                .position(x: frame.midX, y: frame.midY)

                ceremonyText(frame: frame, containerSize: geometry.size)
                    .opacity(titleOpacity * contentOpacity)

                beginButton(frame: frame, containerSize: geometry.size)
                    .opacity(beginOpacity * contentOpacity)
            }
        }
        .ignoresSafeArea()
        .onAppear(perform: start)
    }

    private var showsGlimpsePainting: Bool {
        switch beat {
        case .glimpseBloom, .glimpseHold, .glimpseRecede, .begin, .drain:
            return glimpseAssets != nil
        default:
            return false
        }
    }

    /// Candlelit aubergine breathing into burnt umber — never pure black.
    private var aubergineDarkness: some View {
        ZStack {
            Color(.displayP3, red: 0.16, green: 0.10, blue: 0.17) // deep aubergine
            RadialGradient(
                colors: [
                    Color(.displayP3, red: 0.32, green: 0.18, blue: 0.10).opacity(0.55), // burnt umber warmth
                    Color.clear,
                ],
                center: .center,
                startRadius: 40,
                endRadius: 420
            )
        }
    }

    private func ceremonyText(frame: CGRect, containerSize: CGSize) -> some View {
        VStack(spacing: 14) {
            Text("WELCOME TO LEVEL \(level)")
                .font(.system(size: 15, weight: .semibold, design: .serif))
                .tracking(6)
                .foregroundStyle(Color.ankyGold)

            if !completedTitle.isEmpty {
                Text("“\(completedTitle)” — painted from \(AnkyLevel.thresholdSeconds(forLevel: level).formatted()) seconds of your writing.")
                    .font(.system(size: 14, weight: .regular, design: .serif))
                    .foregroundStyle(Color.ankyPaperDeep.opacity(0.85))
                    .multilineTextAlignment(.center)
                    .padding(.horizontal, 40)
            }
        }
        .position(x: containerSize.width / 2, y: frame.maxY + 64)
    }

    private func beginButton(frame: CGRect, containerSize: CGSize) -> some View {
        Button {
            drainAndFinish()
        } label: {
            Text("Begin")
                .font(.system(size: 16, weight: .medium, design: .serif))
                .foregroundStyle(Color.ankyGoldLight)
                .padding(.horizontal, 38)
                .padding(.vertical, 12)
                .background(
                    Capsule().strokeBorder(Color.ankyGold.opacity(0.55), lineWidth: 1)
                )
        }
        .buttonStyle(.plain)
        .position(x: containerSize.width / 2, y: frame.maxY + 140)
    }

    // MARK: The sequence

    private func start() {
        let packages = coordinator.ceremonyPackages(forLevel: level)
        completedAssets = packages.completed.flatMap(PaintingRevealAssets.init)
        completedTitle = packages.completed?.title ?? ""
        glimpseAssets = packages.glimpse.flatMap(PaintingRevealAssets.init)
        generationExcerpts = coordinator.paintingGenerationExcerpts()
        theme = LevelTheme(package: packages.glimpse ?? packages.completed)

        // 1. The final strokes land. The painting view is inserted in this
        // same render pass, so the animation must start one tick later —
        // otherwise SwiftUI materializes it directly at the end value.
        completedProgress = 0.72
        DispatchQueue.main.async {
            withAnimation(.easeInOut(duration: CeremonyTiming.finalStrokesSeconds)) {
                completedProgress = 1
            }
        }
        after(CeremonyTiming.finalStrokesSeconds) {
            beat = .heldBreath
            after(CeremonyTiming.heldBreathSeconds) {
                beat = .darkening
                withAnimation(.easeInOut(duration: CeremonyTiming.darkeningSeconds)) {
                    darkness = 1
                }
                after(CeremonyTiming.darkeningSeconds) {
                    beat = .title
                    withAnimation(.easeIn(duration: CeremonyTiming.titleFadeSeconds)) {
                        titleOpacity = 1
                    }
                    after(CeremonyTiming.titleFadeSeconds) {
                        beginGlimpse()
                    }
                }
            }
        }
    }

    private func beginGlimpse() {
        if glimpseAssets != nil {
            playGlimpse()
            return
        }
        // The writer outran generation: the darkness simply breathes longer.
        beat = .waitingForGlimpse
        Task {
            if let package = await coordinator.waitForCeremonyPackage(level: level) {
                glimpseAssets = PaintingRevealAssets(package: package)
                theme = LevelTheme(package: package)
            }
            playGlimpse()
        }
    }

    private func playGlimpse() {
        guard glimpseAssets != nil else {
            // Generation is still breathing; don't hold the writer hostage —
            // the glimpse will be waiting on the main painting instead.
            showBegin()
            return
        }
        // The glimpse painting enters this render pass at 0; the 8-second
        // bloom starts on the next tick so the insertion doesn't swallow it.
        beat = .glimpseBloom
        glimpseProgress = 0
        DispatchQueue.main.async {
            withAnimation(.easeInOut(duration: CeremonyTiming.glimpseBloomSeconds)) {
                glimpseProgress = 1
            }
        }
        after(CeremonyTiming.glimpseBloomSeconds) {
            beat = .glimpseHold
            after(CeremonyTiming.glimpseHoldSeconds) {
                beat = .glimpseRecede
                withAnimation(.easeInOut(duration: CeremonyTiming.glimpseRecedeSeconds)) {
                    glimpseProgress = 0
                }
                // Begin fades in during the recede — anky shows you where
                // you're going; you earn it back stroke by stroke.
                showBegin()
            }
        }
    }

    private func showBegin() {
        beat = .begin
        withAnimation(.easeIn(duration: CeremonyTiming.beginFadeSeconds)) {
            beginOpacity = 1
        }
    }

    private func drainAndFinish() {
        beat = .drain
        coordinator.markCeremonyShown(level)
        withAnimation(.easeOut(duration: 0.6)) {
            contentOpacity = 0
        }
        // Pigment bleeds outward into the room: the glow overshoots in the
        // palette, then settles as the darkness lifts.
        withAnimation(.easeInOut(duration: CeremonyTiming.drainSeconds * 0.45)) {
            glowOvershoot = 1.6
        }
        withAnimation(.easeInOut(duration: CeremonyTiming.drainSeconds).delay(CeremonyTiming.drainSeconds * 0.25)) {
            glowOvershoot = 0
            darkness = 0
        }
        after(CeremonyTiming.drainSeconds) {
            onFinished()
        }
    }

    private func after(_ seconds: Double, _ action: @escaping () -> Void) {
        DispatchQueue.main.asyncAfter(deadline: .now() + seconds, execute: action)
    }
}

private struct PaintingGenerationWaitView: View {
    let excerpts: [String]

    var body: some View {
        ZStack {
            WatercolorVeilView(register: .aubergine)

            VStack(spacing: 18) {
                Spacer(minLength: 0)

                Text("Anky is painting...")
                    .font(.system(size: 17, weight: .medium, design: .serif))
                    .foregroundStyle(Color.ankyGoldLight)

                if !excerpts.isEmpty {
                    VStack(spacing: 12) {
                        Text("from your writing")
                            .font(.system(size: 11, weight: .semibold, design: .serif))
                            .textCase(.uppercase)
                            .tracking(2.8)
                            .foregroundStyle(Color.ankyGold.opacity(0.78))

                        VStack(spacing: 10) {
                            ForEach(Array(excerpts.prefix(3).enumerated()), id: \.offset) { index, excerpt in
                                Text("“\(excerpt)”")
                                    .font(.system(size: index == 0 ? 17 : 15, weight: .regular, design: .serif))
                                    .foregroundStyle(Color.ankyPaperDeep.opacity(index == 0 ? 0.92 : 0.72))
                                    .multilineTextAlignment(.center)
                                    .lineSpacing(4)
                                    .fixedSize(horizontal: false, vertical: true)
                                    .padding(.horizontal, 22)
                                    .padding(.vertical, index == 0 ? 15 : 12)
                                    .background(
                                        RoundedRectangle(cornerRadius: 18, style: .continuous)
                                            .fill(Color.ankyInk.opacity(index == 0 ? 0.18 : 0.10))
                                            .overlay(
                                                RoundedRectangle(cornerRadius: 18, style: .continuous)
                                                    .strokeBorder(Color.ankyGold.opacity(index == 0 ? 0.28 : 0.16), lineWidth: 0.6)
                                            )
                                    )
                            }
                        }
                    }
                    .frame(maxWidth: 360)
                }

                Spacer(minLength: 0)
            }
            .padding(.horizontal, 28)
            .padding(.vertical, 56)
        }
        .allowsHitTesting(false)
    }
}

File: apps/android/app/src/main/java/inc/anky/android/feature/painting/UnveilCeremonyView.kt
package inc.anky.android.feature.painting

import androidx.compose.animation.AnimatedVisibility
import androidx.compose.animation.core.Animatable
import androidx.compose.animation.core.CubicBezierEasing
import androidx.compose.animation.core.tween
import androidx.compose.animation.fadeIn
import androidx.compose.animation.fadeOut
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.interaction.MutableInteractionSource
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.BoxWithConstraints
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.offset
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.rememberCoroutineScope
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.alpha
import androidx.compose.ui.draw.drawBehind
import androidx.compose.ui.geometry.Offset
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.colorspace.ColorSpaces
import androidx.compose.ui.semantics.Role
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import inc.anky.android.core.copy.AnkyCopyRegistry
import inc.anky.android.core.level.AnkyLevel
import inc.anky.android.core.level.LevelPaintingCoordinator
import inc.anky.android.core.level.PaintingPackage
import inc.anky.android.ui.lazure.LazurePigments
import inc.anky.android.ui.lazure.LazureWall
import inc.anky.android.ui.lazure.LevelTheme
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.delay
import kotlinx.coroutines.launch
import kotlinx.coroutines.withContext

/**
 * The level-up ceremony.
 *
 * The frame never moves — PaintingFrameMath places it exactly where the
 * main screen holds it; the world darkens and blooms around it. Sequence
 * (CeremonyTiming, second-exact with iOS): final strokes → held breath →
 * candlelit aubergine → WELCOME TO LEVEL {N} → the 8-second glimpse
 * (bloom, breath, recede) → ghost Begin → drain.
 *
 * Every animation is kicked from a LaunchedEffect — never in composition —
 * which is Compose's "one frame after insertion": the freshly inserted
 * painting materializes at its start value before the tween begins (the
 * iOS DispatchQueue.main.async trick, guaranteed by construction here).
 */
@Composable
fun UnveilCeremonyView(
    level: Int,
    coordinator: LevelPaintingCoordinator,
    onFinished: () -> Unit,
) {
    var beat by remember { mutableStateOf(CeremonyBeat.FinalStrokes) }
    var completedPackage by remember { mutableStateOf<PaintingPackage?>(null) }
    var glimpsePackage by remember { mutableStateOf<PaintingPackage?>(null) }
    var generationExcerpts by remember { mutableStateOf<List<String>>(emptyList()) }
    var packagesLoaded by remember { mutableStateOf(false) }

    val completedProgress = remember { Animatable(0.72f) }
    val glimpseProgress = remember { Animatable(0f) }
    val darkness = remember { Animatable(0f) }
    val titleOpacity = remember { Animatable(0f) }
    val beginOpacity = remember { Animatable(0f) }
    val glowOvershoot = remember { Animatable(0f) }
    val contentOpacity = remember { Animatable(1f) }

    val theme = remember(completedPackage, glimpsePackage) {
        (glimpsePackage ?: completedPackage)
            ?.let { LevelTheme.fromPalette(it.palette) }
            ?: LevelTheme.Fallback
    }

    val completedAssets = rememberPaintingRevealAssets(completedPackage)
    val glimpseAssets = rememberPaintingRevealAssets(glimpsePackage)

    val scope = rememberCoroutineScope()
    val easeInOut = remember { CubicBezierEasing(0.42f, 0f, 0.58f, 1f) }
    val easeIn = remember { CubicBezierEasing(0.42f, 0f, 1f, 1f) }

    // MARK: The sequence

    LaunchedEffect(Unit) {
        val (completed, glimpse) = withContext(Dispatchers.IO) { coordinator.ceremonyPackages(level) }
        completedPackage = completed
        glimpsePackage = glimpse
        generationExcerpts = coordinator.paintingGenerationExcerpts()
        packagesLoaded = true

        // 1. The final strokes land: eyes, gold — the 70-80% payoff fires.
        completedProgress.snapTo(0.72f)
        launch {
            completedProgress.animateTo(
                1f,
                tween(CeremonyTiming.millis(CeremonyTiming.FinalStrokesSeconds).toInt(), easing = easeInOut),
            )
        }
        delay(CeremonyTiming.millis(CeremonyTiming.FinalStrokesSeconds))

        // 2. One held breath, fully alive.
        beat = CeremonyBeat.HeldBreath
        delay(CeremonyTiming.millis(CeremonyTiming.HeldBreathSeconds))

        // 3. Candlelit aubergine — living darkness, never pure black.
        beat = CeremonyBeat.Darkening
        launch {
            darkness.animateTo(
                1f,
                tween(CeremonyTiming.millis(CeremonyTiming.DarkeningSeconds).toInt(), easing = easeInOut),
            )
        }
        delay(CeremonyTiming.millis(CeremonyTiming.DarkeningSeconds))

        // 4. WELCOME TO LEVEL {N}.
        beat = CeremonyBeat.Title
        launch {
            titleOpacity.animateTo(
                1f,
                tween(CeremonyTiming.millis(CeremonyTiming.TitleFadeSeconds).toInt(), easing = easeIn),
            )
        }
        delay(CeremonyTiming.millis(CeremonyTiming.TitleFadeSeconds))

        // 5. The glimpse. If the writer outran generation, the darkness
        // simply breathes longer while anky paints.
        if (glimpsePackage == null) {
            beat = CeremonyBeat.WaitingForGlimpse
            coordinator.waitForCeremonyPackage(level)?.let { glimpsePackage = it }
        }
        if (glimpsePackage != null) {
            beat = CeremonyBeat.GlimpseBloom
            glimpseProgress.snapTo(0f)
            launch {
                glimpseProgress.animateTo(
                    1f,
                    tween(CeremonyTiming.millis(CeremonyTiming.GlimpseBloomSeconds).toInt(), easing = easeInOut),
                )
            }
            delay(CeremonyTiming.millis(CeremonyTiming.GlimpseBloomSeconds))

            beat = CeremonyBeat.GlimpseHold
            delay(CeremonyTiming.millis(CeremonyTiming.GlimpseHoldSeconds))

            beat = CeremonyBeat.GlimpseRecede
            launch {
                glimpseProgress.animateTo(
                    0f,
                    tween(CeremonyTiming.millis(CeremonyTiming.GlimpseRecedeSeconds).toInt(), easing = easeInOut),
                )
            }
        }
        // 6. Begin fades in during the recede — anky shows you where you're
        // going; you earn it back stroke by stroke. (Generation still
        // breathing? Don't hold the writer hostage — the glimpse will be
        // waiting on the main painting instead.)
        beat = CeremonyBeat.Begin
        beginOpacity.animateTo(
            1f,
            tween(CeremonyTiming.millis(CeremonyTiming.BeginFadeSeconds).toInt(), easing = easeIn),
        )
    }

    fun drainAndFinish() {
        if (beat == CeremonyBeat.Drain) return
        beat = CeremonyBeat.Drain
        coordinator.markCeremonyShown(level)
        val drainMs = CeremonyTiming.millis(CeremonyTiming.DrainSeconds)
        scope.launch { contentOpacity.animateTo(0f, tween(600)) }
        // Pigment bleeds outward into the room: the glow overshoots in the
        // palette, then settles as the darkness lifts.
        scope.launch {
            glowOvershoot.animateTo(1.6f, tween((drainMs * 0.45).toInt(), easing = easeInOut))
            glowOvershoot.animateTo(0f, tween(drainMs.toInt(), easing = easeInOut))
        }
        scope.launch {
            delay((drainMs * 0.25).toLong())
            darkness.animateTo(0f, tween(drainMs.toInt(), easing = easeInOut))
        }
        scope.launch {
            delay(drainMs)
            onFinished()
        }
    }

    val showsGlimpsePainting = glimpsePackage != null && beat in setOf(
        CeremonyBeat.GlimpseBloom,
        CeremonyBeat.GlimpseHold,
        CeremonyBeat.GlimpseRecede,
        CeremonyBeat.Begin,
        CeremonyBeat.Drain,
    )

    BoxWithConstraints(Modifier.fillMaxSize()) {
        val frame = PaintingFrameMath.frameRect(maxWidth.value, maxHeight.value)

        // The room: warm paper below, living aubergine darkness above.
        LazureWall(mood = theme.wallMood)
        Box(
            Modifier
                .fillMaxSize()
                .alpha(darkness.value)
                .drawBehind {
                    drawRect(CeremonyAubergine)
                    drawRect(
                        Brush.radialGradient(
                            colors = listOf(
                                CeremonyBurntUmber.copy(alpha = 0.55f),
                                CeremonyBurntUmber.copy(alpha = 0f),
                            ),
                            center = Offset(size.width / 2f, size.height / 2f),
                            radius = 420.dp.toPx(),
                        ),
                    )
                },
        )

        AnimatedVisibility(
            visible = beat == CeremonyBeat.WaitingForGlimpse,
            enter = fadeIn(tween(400)),
            exit = fadeOut(tween(400)),
        ) {
            PaintingGenerationWaitView(excerpts = generationExcerpts)
        }

        // The never-moving frame.
        if (packagesLoaded) {
            if (showsGlimpsePainting) {
                PaintingView(
                    assets = glimpseAssets,
                    progress = glimpseProgress.value.toDouble(),
                    glowTint = theme.glowTint,
                    glowStrength = 1f + glowOvershoot.value,
                    modifier = Modifier
                        .offset(x = frame.x.dp, y = frame.y.dp)
                        .size(frame.width.dp, frame.height.dp),
                )
            } else if (completedPackage != null) {
                PaintingView(
                    assets = completedAssets,
                    progress = completedProgress.value.toDouble(),
                    glowTint = theme.glowTint,
                    glowStrength = 1f + glowOvershoot.value,
                    modifier = Modifier
                        .offset(x = frame.x.dp, y = frame.y.dp)
                        .size(frame.width.dp, frame.height.dp),
                )
            }
        }

        // WELCOME TO LEVEL {N} + the painting's provenance line.
        Column(
            horizontalAlignment = Alignment.CenterHorizontally,
            verticalArrangement = Arrangement.spacedBy(14.dp),
            modifier = Modifier
                .fillMaxWidth()
                .offset(y = (frame.maxY + 40f).dp)
                .alpha(titleOpacity.value * contentOpacity.value),
        ) {
            Text(
                text = AnkyCopyRegistry.ceremonyTitle(level),
                style = TextStyle(
                    fontFamily = FontFamily.Serif,
                    fontWeight = FontWeight.SemiBold,
                    fontSize = 15.sp,
                    letterSpacing = 6.sp,
                ),
                color = LazurePigments.ankyGold,
            )
            val completedTitle = completedPackage?.title.orEmpty()
            if (completedTitle.isNotEmpty()) {
                Text(
                    text = AnkyCopyRegistry.ceremonyLine(
                        paintingTitle = completedTitle,
                        seconds = AnkyLevel.thresholdSeconds(level)
                            .coerceAtMost(Int.MAX_VALUE.toLong())
                            .toInt(),
                    ),
                    style = TextStyle(
                        fontFamily = FontFamily.Serif,
                        fontSize = 14.sp,
                        lineHeight = 20.sp,
                        textAlign = TextAlign.Center,
                    ),
                    color = LazurePigments.ankyPaperDeep.copy(alpha = 0.85f),
                    modifier = Modifier.padding(horizontal = 40.dp),
                )
            }
        }

        // The ghost Begin button.
        Box(
            Modifier
                .fillMaxWidth()
                .offset(y = (frame.maxY + 122f).dp)
                .alpha(beginOpacity.value * contentOpacity.value),
            contentAlignment = Alignment.Center,
        ) {
            val interactionSource = remember { MutableInteractionSource() }
            Text(
                text = AnkyCopyRegistry.ceremonyBegin,
                style = TextStyle(
                    fontFamily = FontFamily.Serif,
                    fontWeight = FontWeight.Medium,
                    fontSize = 16.sp,
                ),
                color = LazurePigments.ankyGoldLight,
                modifier = Modifier
                    .border(1.dp, LazurePigments.ankyGold.copy(alpha = 0.55f), CircleShape)
                    .clickable(
                        interactionSource = interactionSource,
                        indication = null,
                        role = Role.Button,
                        enabled = beat == CeremonyBeat.Begin,
                        onClick = { drainAndFinish() },
                    )
                    .padding(horizontal = 38.dp, vertical = 12.dp),
            )
        }
    }
}

/** Deep aubergine — candlelit darkness, never pure black. */
private val CeremonyAubergine =
    Color(red = 0.16f, green = 0.10f, blue = 0.17f, alpha = 1f, colorSpace = ColorSpaces.DisplayP3)

/** Burnt umber warmth breathing at the center of the darkness. */
private val CeremonyBurntUmber =
    Color(red = 0.32f, green = 0.18f, blue = 0.10f, alpha = 1f, colorSpace = ColorSpaces.DisplayP3)
